service/udp.go

```diff
@@ -127,97 +132,133 @@ func (h *packetHandler) SetTargetIPValidator(targetIPValidator onet.TargetIPVali
 	h.targetIPValidator = targetIPValidator
 }
 
-// Listen on addr for encrypted packets and basically do UDP NAT.
-// We take the ciphers as a pointer because it gets replaced on config updates.
-func (h *packetHandler) Handle(clientConn net.PacketConn) {
-	nm := newNATmap(h.natTimeout, h.m, h.logger)
-	defer nm.Close()
-	cipherBuf := make([]byte, serverUDPBufferSize)
-	textBuf := make([]byte, serverUDPBufferSize)
+type PacketHandleFunc func(conn net.Conn, pkt []byte)
```

Comment:
We need to introduce a NAT component that reads a packet and sends to the right association, to provide the functionality we get in Caddy.

Then both servers would call a UDP Association handler. That would be the abstraction that is shared.